packages/expo-sqlite/src/SQLite.ts

```diff
@@ -35,12 +36,12 @@ export class SQLiteDatabase {
   /**
    * Executes the SQL statement and returns a callback resolving with the result.
    */
-  exec(queries: Query[], readOnly: boolean, callback: SQLiteCallback): void {
+  exec(queries: Query[], readOnly: boolean, requiresSync: boolean, callback: SQLiteCallback): void {
```

Comment:
imo you guys would want to figure out some way to remove any flags related to `requiresSync`

One option is to use two separate connections for writes. One for writes from the sync layer, one for writes from the app itself. This way you can differentiate between remote and local events and it can be hidden away from developers that don't need to know those details but still available for those that do.